pkg/asset/agent/installconfig.go

```diff
@@ -187,58 +188,53 @@ func warnUnusedConfig(installConfig *types.InstallConfig) {
 	switch installConfig.Platform.Name() {
 
 	case baremetal.Name:
+		default_ic := &types.InstallConfig{Platform: types.Platform{BareMetal: &baremetal.Platform{}}}
+		baremetaldefaults.SetPlatformDefaults(default_ic.Platform.BareMetal, default_ic)
+
 		baremetal := installConfig.Platform.BareMetal
-		// +kubebuilder:default="qemu:///system". Set from generic install config code
-		if baremetal.LibvirtURI != "qemu:///system" {
+		defaultBM := default_ic.Platform.BareMetal
+		// Compare values from generic installconfig code to check for changes
+		if baremetal.LibvirtURI != defaultBM.LibvirtURI {
 			fieldPath := field.NewPath("Platform", "Baremetal", "LibvirtURI")
 			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.LibvirtURI))
 		}
-		if baremetal.ClusterProvisioningIP != "" {
+		if baremetal.ClusterProvisioningIP != defaultBM.ClusterProvisioningIP {
 			fieldPath := field.NewPath("Platform", "Baremetal", "ClusterProvisioningIP")
 			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.ClusterProvisioningIP))
 		}
-		if baremetal.DeprecatedProvisioningHostIP != "" {
+		if baremetal.DeprecatedProvisioningHostIP != defaultBM.DeprecatedProvisioningHostIP {
 			fieldPath := field.NewPath("Platform", "Baremetal", "ProvisioningHostIP")
 			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.DeprecatedProvisioningHostIP))
 		}
-		if baremetal.BootstrapProvisioningIP != "" {
+		if baremetal.BootstrapProvisioningIP != defaultBM.BootstrapProvisioningIP {
 			fieldPath := field.NewPath("Platform", "Baremetal", "BootstrapProvisioningIP")
 			logrus.Debugf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.BootstrapProvisioningIP))
 		}
-		if baremetal.ExternalBridge != "" {
+		if baremetal.ExternalBridge != defaultBM.ExternalBridge {
 			fieldPath := field.NewPath("Platform", "Baremetal", "ExternalBridge")
 			logrus.Warnf(fmt.Sprintf("%s: %s is ignored", fieldPath, baremetal.ExternalBridge))
 		}
-		if baremetal.ExternalMACAddress != "" {
```

Comment:
Good catch. I would have assumed the MAC address would have been blank instead of randomized.